schema/appState.json

```diff
@@ -0,0 +1,390 @@
+                    "oneOf": [
+                        {
+                            "type": "null"
+                        },
+                        {
+                            "type": "object",
+                            "properties": {
+                                "variables": {
+                                    "$ref": "#/definitions/StringArray"
+                                },
+                                "parameters": {
+                                    "type": "array",
+                                    "items": {
+                                        "$ref": "#/definitions/OdinParameter"
+                                    }
+                                },
+                                "messages": {
+                                    "$ref": "#/definitions/StringArray"
+                                }
+                            },
+                            "required": ["variables", "parameters", "messages"],
+                            "additionalProperties": false
+                        }
+                    ]
+                },
+                "model": {
+                    "type": ["null", "string"]
+                },
+                "error": {
+                    "type": {
+                        "oneOf": [
+                            { "type": "null" },
+                            { "$ref": "#definitions/OdinModelResponseError" }
+                        ]
+                    }
+                }
+            },
+            "requiredProperties": ["metadata", "model", "error"],
+            "additionalProperties": false
+        },
+
+        "OdinParameter": {
+            "type": "object",
+            "properties": {
+                "name": {
+                    "type": "string"
+                },
+                "default": {
+                    "type": ["null", "number"]
+                },
+                "min": {
+                    "type": ["null", "number"]
+                },
+                "max": {
+                    "type": ["null", "number"]
+                },
+                "is_integer": {
+                    "type": "boolean"
+                },
+                "rank": {
+                    "type": "number"
+                }
+            },
+            "additionalProperties": false,
+            "required": ["name", "default", "min", "max", "is_integer", "rank"]
+        },
+
+        "OdinModelResponseError": {
+            "type": "object",
+            "properties": {
+                "line": {
+                    "comment": "This is surprising to me, but matches the current type - I'd have thought this would be number[]",
+                    "$ref": "#/definitions/StringArray"
+                },
+                "message": {
+                    "type": "string"
+                }
+            },
+            "required": [],
+            "additionalProperties": false
+        },
+
+        "OdinSolution": {
```

Comment:
```suggestion
        "OdinSolutionInputs": {
```
?

Reply:
Probably that's the right name, yeah. It will hopefully be clearer once we get the code change through to support this

Reply:
this should be clearer now